NotificationService/RLTNotificationsTracker.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  RLTNotificationsTracker.swift
+//  RealifeTech-SDK
+//
+//  Created by YOU-HSUAN YU on 2022/6/27.
+//
+
+import UserNotifications
+
+public class RLTNotificationsTracker {
+
+    private let appGroupStore: AppGroupUserDefaultsStore?
+
+    public init(appGroupId: String) {
+        self.appGroupStore = AppGroupUserDefaultsStore(appGroupId: appGroupId)
+    }
+
+    public func didReceive(_ request: UNNotificationRequest,
+                           withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
+        if let bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent),
+           let userInfo = bestAttemptContent.userInfo as? [String: Any] {
+            configureSDKAndRegisterDevice {
+                RealifeTech.Communicate?.trackPush(event: .received, trackInfo: userInfo) { _ in
```

Comment:
Should communicate be passed as a dependency? Or not possible because the integrator initialises this class?

Reply:
That's correct. I'm afraid it's not possible as integrator initialises this class.